x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -192,6 +195,11 @@
 
     private volatile long lastExpirationRunMs;
 
+    private static final long EVICTION_CHECKING_INTERVAL = 600_000L; // 600K ms = 10 minutes
+    private static final long EVICTION_WARNING_THRESHOLD = 15 * EVICTION_CHECKING_INTERVAL / 1000; // 15 eviction per sec = 9000 in 10 min
```

Comment:
We can tweak this later - please don't hold up the PR while we discuss it, but I think this number could be too high.

9,000 in 10 minutes is certainly thrashing, but I would think it's an issue at a much smaller number as well.
If 1,000 items get evicted (due to cache size) over a 10 minute period, isn't that bad?

I don't know what the best heuristic is, but I don't think it needs to be a sustained rate of 15 per second for the full 10 minutes to count as a problem.

Reply:
Yes this threshold is very liberal. A small node is probably tripped over before the warning is issued. I lowered it to 5 min. We can further tweak it. But I also don't want it to be purely based on the smallest node. For a large deployment, assuming 3 nodes each with multiple cores, this threshold might be OKish?